From af97bbc282614c8c4af250f010626a0019742ef2 Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Tue, 10 Jun 2014 10:21:25 +0300
Subject: [PATCH 190/213] Version 0.9.9.7: Mobile numbers when registering SIM
 card

Now checking wheter the old and new mobile numbers when switching
numbers are blank
---
 .../org/cgiar/ilri/mistro/farmer/LandingActivity.java   | 17 ++++++++++++-----
 MistroFarmer/src/main/res/values/strings_en.xml         |  2 ++
 MistroFarmer/src/main/res/values/strings_kp.xml         |  2 ++
 MistroFarmer/src/main/res/values/strings_kr.xml         |  2 ++
 MistroFarmer/src/main/res/values/strings_lu.xml         |  2 ++
 MistroFarmer/src/main/res/values/strings_nn.xml         |  2 ++
 MistroFarmer/src/main/res/values/strings_sw.xml         |  2 ++
 7 files changed, 24 insertions(+), 5 deletions(-)

diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
index d9c18f2..dfa4a7d 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
@@ -276,11 +276,18 @@ public class LandingActivity extends SherlockActivity implements MistroActivity,
 
     private void registerSimCard()
     {
-        TelephonyManager telephonyManager=(TelephonyManager)this.getSystemService(Context.TELEPHONY_SERVICE);
-        if(telephonyManager.getSimSerialNumber()!=null)
-        {
-            SimCardRegistrationThread simCardRegistrationThread=new SimCardRegistrationThread();
-            simCardRegistrationThread.execute(oldMobileNumberET.getText().toString(),newMobileNumberET.getText().toString(),telephonyManager.getSimSerialNumber());
+        if(oldMobileNumberET.getText().toString().length() == 0){
+            Toast.makeText(this, Locale.getStringInLocale("enter_old_phone_no", this), Toast.LENGTH_LONG).show();
+        }
+        else if(newMobileNumberET.getText().toString().length() == 0){
+            Toast.makeText(this, Locale.getStringInLocale("enter_new_phone_no", this), Toast.LENGTH_LONG).show();
+        }
+        else{
+            TelephonyManager telephonyManager=(TelephonyManager)this.getSystemService(Context.TELEPHONY_SERVICE);
+            if(telephonyManager.getSimSerialNumber()!=null) {
+                SimCardRegistrationThread simCardRegistrationThread=new SimCardRegistrationThread();
+                simCardRegistrationThread.execute(oldMobileNumberET.getText().toString(),newMobileNumberET.getText().toString(),telephonyManager.getSimSerialNumber());
+            }
         }
     }
 
diff --git a/MistroFarmer/src/main/res/values/strings_en.xml b/MistroFarmer/src/main/res/values/strings_en.xml
index fc7e747..fddba32 100644
--- a/MistroFarmer/src/main/res/values/strings_en.xml
+++ b/MistroFarmer/src/main/res/values/strings_en.xml
@@ -309,4 +309,6 @@
     <string name="epersonnel_tshoot_en">Unable to fetch extension personnel from the server. Try pressing back and restarting the registration</string>
     <string name="enter_two_names_en">Enter at least two names</string>
     <string name="phone_number_not_valid_en">The phone number you entered is not valid</string>
+    <string name="enter_old_phone_no_en">Please enter your old phone number</string>
+    <string name="enter_new_phone_no_en">Please enter your new phone number</string>
 </resources>
diff --git a/MistroFarmer/src/main/res/values/strings_kp.xml b/MistroFarmer/src/main/res/values/strings_kp.xml
index 28599e9..97892f4 100644
--- a/MistroFarmer/src/main/res/values/strings_kp.xml
+++ b/MistroFarmer/src/main/res/values/strings_kp.xml
@@ -307,4 +307,6 @@
     <string name="epersonnel_tshoot_kp">Unable to fetch extension personnel from the server. Try pressing back and restarting the registration</string>
     <string name="enter_two_names_kp">Enter at least two names</string>
     <string name="phone_number_not_valid_kp">The phone number you entered is not valid</string>
+    <string name="enter_old_phone_no_kp">Please enter your old phone number</string>
+    <string name="enter_new_phone_no_kp">Please enter your new phone number</string>
 </resources>
diff --git a/MistroFarmer/src/main/res/values/strings_kr.xml b/MistroFarmer/src/main/res/values/strings_kr.xml
index 37f8532..4af0235 100644
--- a/MistroFarmer/src/main/res/values/strings_kr.xml
+++ b/MistroFarmer/src/main/res/values/strings_kr.xml
@@ -306,4 +306,6 @@
     <string name="epersonnel_tshoot_kr">Unable to fetch extension personnel from the server. Try pressing back and restarting the registration</string>
     <string name="enter_two_names_kr">Enter at least two names</string>
     <string name="phone_number_not_valid_kr">The phone number you entered is not valid</string>
+    <string name="enter_old_phone_no_kr">Please enter your old phone number</string>
+    <string name="enter_new_phone_no_kr">Please enter your new phone number</string>
 </resources>
\ No newline at end of file
diff --git a/MistroFarmer/src/main/res/values/strings_lu.xml b/MistroFarmer/src/main/res/values/strings_lu.xml
index b4ff1ed..aa70b94 100644
--- a/MistroFarmer/src/main/res/values/strings_lu.xml
+++ b/MistroFarmer/src/main/res/values/strings_lu.xml
@@ -309,4 +309,6 @@
     <string name="epersonnel_tshoot_lu">Unable to fetch extension personnel from the server. Try pressing back and restarting the registration</string>
     <string name="enter_two_names_lu">Enter at least two names</string>
     <string name="phone_number_not_valid_lu">The phone number you entered is not valid</string>
+    <string name="enter_old_phone_no_lu">Please enter your old phone number</string>
+    <string name="enter_new_phone_no_lu">Please enter your new phone number</string>
 </resources>
diff --git a/MistroFarmer/src/main/res/values/strings_nn.xml b/MistroFarmer/src/main/res/values/strings_nn.xml
index 7f01879..0464f3f 100644
--- a/MistroFarmer/src/main/res/values/strings_nn.xml
+++ b/MistroFarmer/src/main/res/values/strings_nn.xml
@@ -307,4 +307,6 @@
     <string name="epersonnel_tshoot_nn">Unable to fetch extension personnel from the server. Try pressing back and restarting the registration</string>
     <string name="enter_two_names_nn">Enter at least two names</string>
     <string name="phone_number_not_valid_nn">The phone number you entered is not valid</string>
+    <string name="enter_old_phone_no_nn">Please enter your old phone number</string>
+    <string name="enter_new_phone_no_nn">Please enter your new phone number</string>
 </resources>
diff --git a/MistroFarmer/src/main/res/values/strings_sw.xml b/MistroFarmer/src/main/res/values/strings_sw.xml
index dc5f102..11a34e4 100644
--- a/MistroFarmer/src/main/res/values/strings_sw.xml
+++ b/MistroFarmer/src/main/res/values/strings_sw.xml
@@ -307,4 +307,6 @@
     <string name="epersonnel_tshoot_sw">Unable to fetch extension personnel from the server. Try pressing back and restarting the registration</string>
     <string name="enter_two_names_sw">Enter at least two names</string>
     <string name="phone_number_not_valid_sw">The phone number you entered is not valid</string>
+    <string name="enter_old_phone_no_sw">Please enter your old phone number</string>
+    <string name="enter_new_phone_no_sw">Please enter your new phone number</string>
 </resources>
-- 
2.7.1

